tests/ui/entry.rs

```diff
@@ -68,4 +68,11 @@ fn insert_other_if_absent<K: Eq + Hash, V>(m: &mut HashMap<K, V>, k: K, o: K, v:
     }
 }
 
+// should not trigger, because the one uses different HashMap from another one
+fn insert_other<K: Eq + Hash, V>(m: &mut HashMap<K, V>, n: &mut HashMap<K, V>, k: K, v: V) {
+    if !m.contains_key(&k) {
+        n.insert(k, v);
+    }
+}
+
```

Comment:
Could you also add the following test as described in the issue (where one is mut and the other is not):

```rust
fn insert_other<K: Eq + Hash, V>(m: HashMap<K, V>, n: &mut HashMap<K, V>, k: K, v: V) {
    if !m.contains_key(&k) {
        n.insert(k, v);
    }
}
```

Reply:
added the test (and renamed)